From d58fba9683a926f1230e4f5f574b083bc37f8b07 Mon Sep 17 00:00:00 2001
From: Ong Boon Leong <boon.leong.ong@intel.com>
Date: Sat, 21 Sep 2019 15:27:55 +0800
Subject: [PATCH 05/17] net: stmmac: call stmmac_clear_tx_descriptors() in
 stmmac_tx_err()

This reduces code duplication and prepares for better XDP setup in future.

Signed-off-by: Ong Boon Leong <boon.leong.ong@intel.com>
---
 drivers/net/ethernet/stmicro/stmmac/stmmac_main.c | 15 +--------------
 1 file changed, 1 insertion(+), 14 deletions(-)

diff --git a/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c b/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
index 81f121dd8ae0..72dfe6d7624e 100644
--- a/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
+++ b/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
@@ -2122,7 +2122,6 @@ static int stmmac_tx_clean(struct stmmac_priv *priv, int budget, u32 queue)
 static void stmmac_tx_err(struct stmmac_priv *priv, u32 chan)
 {
 	struct stmmac_tx_queue *tx_q = &priv->tx_queue[chan];
-	int i;
 
 	netif_tx_stop_queue(netdev_get_tx_queue(priv->dev, chan));
 
@@ -2130,19 +2129,7 @@ static void stmmac_tx_err(struct stmmac_priv *priv, u32 chan)
 	stmmac_stop_mac_tx(priv, priv->ioaddr);
 
 	dma_free_tx_skbufs(priv, chan);
-	for (i = 0; i < priv->dma_tx_size; i++)
-		if (priv->extend_desc)
-			stmmac_init_tx_desc(priv, &tx_q->dma_etx[i].basic,
-					    priv->mode,
-					    (i == priv->dma_tx_size - 1));
-		else if (priv->enhanced_tx_desc)
-			stmmac_init_tx_desc(priv, &tx_q->dma_enhtx[i].basic,
-					    priv->mode,
-					    (i == priv->dma_tx_size - 1));
-		else
-			stmmac_init_tx_desc(priv, &tx_q->dma_tx[i],
-					    priv->mode,
-					    (i == priv->dma_tx_size - 1));
+	stmmac_clear_tx_descriptors(priv, chan);
 	tx_q->dirty_tx = 0;
 	tx_q->cur_tx = 0;
 	tx_q->mss = 0;
-- 
2.17.0

